resolver/resolver.go

```diff
@@ -581,69 +577,54 @@ func (res *Resolver) RestHost(req *restful.Request, resp *restful.Response) {
 		logging.CurLog.NonMesosRequests.Inc()
 		logging.CurLog.NonMesosFailed.Inc()
 	}
-
 }
 
 // Reports Mesos-DNS version through http interface
 func (res *Resolver) RestPorts(req *restful.Request, resp *restful.Response) {
-	io.WriteString(resp, "To be implemented...")
+	err := resp.WriteErrorString(http.StatusNotImplemented, "To be implemented...")
+	if err != nil {
+		logging.Error.Println(err)
+	}
 }
 
 // Reports Mesos-DNS version through http interface
 func (res *Resolver) RestService(req *restful.Request, resp *restful.Response) {
-
-	var ip string
-
-	service := req.PathParameter("service")
-
+	svc := req.PathParameter("service")
 	// clean up service name
-	dom := strings.ToLower(cleanWild(service))
+	dom := strings.ToLower(cleanWild(svc))
 	if dom[len(dom)-1] != '.' {
 		dom += "."
 	}
-
-	mapS := make([]map[string]string, 0)
 	rs := res.records()
 
-	for _, srv := range rs.SRVs[dom] {
-		h, port, _ := net.SplitHostPort(srv)
-		p, _ := strconv.Atoi(port)
+	type service struct {
+		Service string `json:"service"`
+		Host    string `json:"host"`
+		IP      string `json:"ip"`
+		Port    string `json:"port"`
+	}
+
+	services := make([]service, 0, len(rs.SRVs[dom]))
```

Comment:
I don't know, but I'm not worried about that here at all. Do you think I should?